src/System.Text.Json/src/System/Text/Json/JsonDocument.TryGetProperty.cs

```diff
@@ -0,0 +1,232 @@
+                        utf8Name.Clear();
+                        ArrayPool<byte>.Shared.Return(tmpUtf8);
+                    }
+                }
+
+                // Move to the previous value
+                candidateIndex -= DbRow.Size;
+            }
+
+            // None of the property names were within the range that the UTF-8 encoding would have been.
+            value = default;
+            return false;
+        }
+
+        internal bool TryGetNamedPropertyValue(int index, ReadOnlySpan<byte> propertyName, out JsonElement value)
+        {
+            CheckNotDisposed();
+
+            DbRow row = _parsedData.Get(index);
+
+            CheckExpectedType(JsonTokenType.StartObject, row.TokenType);
+
+            // Only one row means it was EndObject.
+            if (row.NumberOfRows == 1)
+            {
+                value = default;
+                return false;
+            }
+
+            int endIndex = checked(row.NumberOfRows * DbRow.Size + index);
+
+            return TryGetNamedPropertyValue(
+                index + DbRow.Size,
+                endIndex,
+                propertyName,
+                out value);
+        }
+
+        private bool TryGetNamedPropertyValue(
+            int startIndex,
+            int endIndex,
+            ReadOnlySpan<byte> propertyName,
+            out JsonElement value)
+        {
+            DbRow row;
+            ReadOnlySpan<byte> documentSpan = _utf8Json.Span;
+
+            // Move to the row before the EndObject
+            int index = endIndex - DbRow.Size;
+
+            while (index > startIndex)
+            {
+                row = _parsedData.Get(index);
+                Debug.Assert(row.TokenType != JsonTokenType.PropertyName);
+
+                // Move before the value
+                if (row.IsSimpleValue)
+                {
+                    index -= DbRow.Size;
+                }
+                else
+                {
+                    Debug.Assert(row.NumberOfRows > 0);
+                    index -= DbRow.Size * (row.NumberOfRows + 1);
+                }
+
+                row = _parsedData.Get(index);
+                Debug.Assert(row.TokenType == JsonTokenType.PropertyName);
+
+                ReadOnlySpan<byte> currentPropertyName = documentSpan.Slice(row.Location, row.SizeOrLength);
+
+                if (row.HasComplexChildren)
+                {
+                    // An escaped property name will be longer than an unescaped candidate, so only unescape
+                    // when the lengths are compatible.
+                    if (currentPropertyName.Length > propertyName.Length)
+                    {
+                        int idx = currentPropertyName.IndexOf(JsonConstants.BackSlash);
+                        Debug.Assert(idx >= 0);
+
+                        // If everything up to where the property name has a backslash matches, keep going.
+                        if (propertyName.Length > idx &&
```

Comment:
Why `> idx` and not `>= idx`?

Reply:
`"thisIsEscaped\u0020"` vs `"thisIsEscaped"`.

If the property name is only `idx` long then it ended where the candidate started escaping, therefore it won't match.